core/services/head_tracker.go

```diff
@@ -460,10 +460,10 @@ func (ht *HeadTracker) receiveHeaders(ctx context.Context) error {
 
 				err := ht.handleNewHead(ctx, blockHeader)
 				if ctx.Err() != nil {
-					return nil
+					return ctx.Err()
 				} else if deadlineCtx.Err() != nil {
 					ht.logger().Warnw("HeadTracker: handling of new head timed out", "error", ctx.Err(), "timeBudget", timeBudget.String())
-					return err
+					return deadlineCtx.Err()
```

Comment:
I think we just want to continue here - if we return an error then we'll log an (incorrect) error saying we are unsubscribing and then resubscribe which doesn't make sense for just slow app code processing of a head.

Reply:
yeah good point